src/python/foglamp/configuration_manager.py

```diff
@@ -0,0 +1,370 @@
+        FROM 
+            foglamp.configuration
+        WHERE 
+            configuration.key='{}'
+    """
+    query_full = query_template.format(category_item, category_key)
+    try:
+        async with aiopg.sa.create_engine('postgresql://foglamp:foglamp@localhost:5432/foglamp') as engine:
+            async with engine.acquire() as conn:
+                async for row in conn.execute(text(query_full).columns(_configuration_tbl.c.value)):
+                    return row.value
+    except Exception:
+        logging.getLogger('configuration-manager').exception('Unable to read all category items')
+        print("error")
+    return None
+
+
+async def _read_category_item_value(category_key, category_item):
+    query_template = """
+        SELECT 
+            configuration.value::json->'{}'->'Value' as value
+        FROM 
+            foglamp.configuration
+        WHERE 
+            configuration.key='{}'
+    """
+    query_full = query_template.format(category_item, category_key)
+    try:
+        async with aiopg.sa.create_engine('postgresql://foglamp:foglamp@localhost:5432/foglamp') as engine:
+            async with engine.acquire() as conn:
+                async for row in conn.execute(text(query_full).columns(_configuration_tbl.c.value)):
+                    return row.value
+    except Exception:
+        logging.getLogger('configuration-manager').exception('Unable to read all category items')
+        print("error")
+    return None
+
+
+async def _update_item_value(category_key, category_item, item_value_replacement):
+    query_template = """
+            UPDATE foglamp.configuration 
+            SET value = jsonb_set(value, '{{{},Value}}', '"{}"') 
+            WHERE key='{}'
+        """
+    query_full = query_template.format(category_item, item_value_replacement, category_key)
+    # print(query_full)
+
+    try:
+        async with aiopg.sa.create_engine('postgresql://foglamp:foglamp@localhost:5432/foglamp') as engine:
+            async with engine.acquire() as conn:
+                await conn.execute(query_full)
+    except Exception:
+        logging.getLogger('configuration-manager').exception(
+            'Unable to update category_name (%s) category_item (%s) with new Value (%s)', category_key, category_item,
+            item_value_replacement)
+        print("error")
+
+
+async def get_all_category_names():
+    """Get all category names in the FogLAMP system
+
+    Keyword Arguments:
+    None
+
+    Return Values:
+    a list of strings, each representing a unique category name
+    None
+
+    Side Effects:
+    None
+
+    Exceptions Raised:
+    Unknown
+
+    Restrictions and Usage:
+    A FogLAMP component calls this method to obtain all category names in the FogLamp system
+    """
+    try:
+        return await _read_all_category_keys()
+    except:
+        logging.getLogger('configuration-manager').exception(
+            'Unable to read all category names')
```

Comment:
Logging already happens in _read_all_category_keys, we should not repeat logging at all levels.